Plug-Ins/NovaTools/System Editor/SystemWindowController.swift

```diff
@@ -0,0 +1,335 @@
+import AppKit
+import RFSupport
+import OrderedCollections
+
+class SystemWindowController: AbstractEditor, ResourceEditor {
+    static let supportedTypes = ["sÿsm"]
+    let resource: Resource
+    let manager: RFEditorManager
+
+    @IBOutlet var stellarTable: NSTableView!
+    @IBOutlet var systemView: SystemMapView!
+    private(set) var stellarViews: OrderedDictionary<Int, StellarView> = [:]
+    private var stellarList: [Resource?] = Array(repeating: nil, count: 16)
+    private var isSelectingStellars = false
+
+    required init(resource: Resource, manager: RFEditorManager) {
+        self.manager = manager
+        self.resource = manager.findResource(type: ResourceType("sÿst"), id: resource.id, currentDocumentOnly: true)!
+        super.init(window: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func saveResource(_ sender: Any) {}
+    func revertResource(_ sender: Any) {}
+}
+
+extension SystemWindowController {
+    override var windowNibName: String {
+        "SystemWindow"
+    }
+    var windowTitle: String {
+        "System Map"
+    }
+
+    override func windowDidLoad() {
+        NotificationCenter.default.addObserver(self, selector: #selector(resourceListChanged(_:)), name: .DocumentDidAddResources, object: manager.document)
+        NotificationCenter.default.addObserver(self, selector: #selector(resourceListChanged(_:)), name: .DocumentDidRemoveResources, object: manager.document)
+        NotificationCenter.default.addObserver(self, selector: #selector(resourceIDChanged(_:)), name: .ResourceIDDidChange, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(resourceNameChanged(_:)), name: .ResourceNameDidChange, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(resourceDataChanged(_:)), name: .ResourceDataDidChange, object: nil)
+
+        // Scroll to system center
+        if var documentRect = systemView.enclosingScrollView?.documentVisibleRect {
+            documentRect.origin.y *= -1 // Workaround an issue where the clip view is initially misplaced
+            systemView.scroll(NSPoint(x: systemView.frame.midX - documentRect.midX, y: systemView.frame.midY - documentRect.midY))
+        }
+
+        self.reload()
+    }
+
+    private func reload() {
+        try? read()
+
+        stellarViews = [:]
+        for stellar in stellarList {
+            if stellar != nil && stellarViews[stellar!.id] == nil {
+                self.read(stellar: stellar!)
```

Comment:
Another forced unwrap that can be avoided.
```suggestion
            if let stellar && stellarViews[stellar.id] == nil {
                self.read(stellar: stellar)
```